src/mgr/Registration.cc

```diff
@@ -10,23 +10,71 @@
 
 #include "squid.h"
 #include "CacheManager.h"
+#include "mgr/FunAction.h"
 #include "mgr/Registration.h"
 
+namespace Mgr {
+
+/// creates FunAction using ActionCreator API
+class FunActionCreator: public ActionCreator
+{
+public:
+    explicit FunActionCreator(OBJH *aHandler): handler(aHandler) {}
+
+    /* ActionCreator API */
+    Action::Pointer create(const CommandPointer &cmd) const override {
+        return FunAction::Create(cmd, handler);
+    }
+
+private:
+    OBJH *handler; ///< legacy function to pass to the FunAction wrapper
+};
+
+/// creates Action using supplied Action::Create method and command
+class ClassActionCreator: public ActionCreator
+{
+public:
+    using Handler = ClassActionCreationHandler;
+
+public:
+    ClassActionCreator(Handler *aHandler): handler(aHandler) {}
+
+    /* ActionCreator API */
+    Action::Pointer create(const Command::Pointer &cmd) const override {
+        return handler(cmd);
+    }
+
+private:
+    Handler *handler; ///< configured Action object creator
+};
+
+} // namespace Mgr
```

Comment:
The above code is not new. I moved it here, essentially without changes (I added some missing descriptions and `const` IIRC) to avoid exposing RegisterAction() callers to Mgr code they do not care about. After that move, further code improvements, especially the addition of mgr/ActionFeatures.h header, probably made that move unnecessary and, hence, out-of-scope. Feel free to undo this move or let me know if I should undo it.